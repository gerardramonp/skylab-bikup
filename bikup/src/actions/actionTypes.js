export default {
    LOAD_BIKES: 'LOAD_BIKES',
    LOAD_BIKE_BY_ID: 'LOAD_BIKE_BY_ID'
};
