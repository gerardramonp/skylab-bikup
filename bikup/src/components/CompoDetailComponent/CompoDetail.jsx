import React, { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import bikeStore from '../../stores/bikeStore';

import StandardAside from '../StandardAside/StandardAside';
import DetailCompoCard from './DetailCompoCardComponent/DetailCompoCard';

import './CompoDetail.scss';

function CompoDetail(props) {
    const [compoInfo] = useState(JSON.parse(sessionStorage.actualCompo));
    const [bikeInfo] = useState(JSON.parse(sessionStorage.actualBike));

    return (
        <div className="general-container">
            <div className="desktop">
                <StandardAside />
            </div>

            <div className="compodetail">
                <div className="compodetail__upper mobile">
                    <NavLink to="/bikes/">Back</NavLink>
                    <p className="upper__edit">Edit</p>
                </div>

                <div className="compodetail__head">
                    <div className="head__container">
                        <h2 className="head__compoName">
                            {bikeInfo.bikeName} - {compoInfo.compoDisplayName}
                        </h2>
                        <button className="compodetail__reset desktop">
                            Reset Component
                        </button>
                    </div>
                    <div className="separator"></div>

                    <div className="compodetail__km-hours">
                        <div className="km-hours km-hours__km">
                            <p className="km-hours__title">Total KM</p>
                            <div className="separator-small"></div>
                            <p className="km__value">
                                {compoInfo.compoAccumulatedMeters / 1000}
                            </p>
                        </div>
                        <div className="km-hours__hours km-hours">
                            <p className="km-hours__title">Total Hours</p>
                            <div className="separator-small"></div>
                            <p className="km-hours__value">
                                {Math.floor(
                                    compoInfo.compoAccumulatedMinutes / 60
                                )}
                            </p>
                        </div>
                    </div>
                </div>

                <div className="compodetail__components">
                    <DetailCompoCard compoInfo={compoInfo} />
                </div>

                <div className="compodetail__details">
                    <div className="details__content">
                        <img
                            src="https://image.flaticon.com/icons/svg/2623/2623442.svg"
                            alt="bike type"
                            className="details__type-img"
                        />
                        <div className="details__info">
                            <p className="info__type">
                                {bikeInfo.bikeName} -{' '}
                                {compoInfo.compoDisplayName}
                            </p>
                            <p>'Unknown Brand' - 'Unknown Model'</p>
                            <p>Someinfo</p>
                        </div>
                    </div>
                </div>
                <button className="compodetail__reset mobile">
                    Reset Component
                </button>
            </div>
        </div>
    );
}

export default CompoDetail;
