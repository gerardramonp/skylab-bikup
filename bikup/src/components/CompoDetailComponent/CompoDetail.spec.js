import React from 'react';
import renderer from 'react-test-renderer';
import CompoDetail from './CompoDetail';
import { BrowserRouter as Router } from 'react-router-dom';
import dispatcher from '../../dispatcher';
import actionTypes from '../../actions/actionTypes';

describe('CompoDetail Component', () => {
	let tree = null;

	beforeEach(() => {
		tree = renderer.create(
			<Router>
				<CompoDetail props={'hola'} />
			</Router>
		);
	});

	it('should match', () => {
		tree = renderer.create(
			<Router>
				<CompoDetail props={'hola'} />
			</Router>
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it('Should call onChange function', () => {
		dispatcher.dispatch({
			type: actionTypes.LOAD_BIKE_BY_ID,
			data: 'random',
		});

		expect(tree).toBeTruthy();
	});
});
