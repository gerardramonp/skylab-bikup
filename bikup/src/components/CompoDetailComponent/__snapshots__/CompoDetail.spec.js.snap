// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompoDetail Component should match 1`] = `
Array [
  <div
    className="mobile__menu"
  >
    <div
      className="menu__decorator"
      onClick={[Function]}
    >
      <div
        className="decorator__line"
      />
    </div>
    <ul
      className="menu__list"
    >
      <li
        className="menu__option"
        onClick={[Function]}
      >
        <img
          alt="Logout"
          className="option__img"
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f5a496d520b7b130a5d4819/7f0d35433cdacef755bec9d18cbcce2c/logout.png"
        />
        <p
          className="option__p"
        >
          Logout
        </p>
      </li>
    </ul>
  </div>,
  <header
    className="header"
  >
    <div
      className="header__mobile"
    >
      <div
        className="mobile__nav"
      >
        <img
          alt="logo"
          className="logo"
          src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
        />
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          <img
            alt="ExploreMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/214746d15b751953f02eb252fb879874/magnifying-glass.png"
          />
        </a>
        <a
          aria-current={null}
          href="/bikes"
          onClick={[Function]}
        >
          <img
            alt="MyBikesMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ca1db79d00b304cb995a7491f0ce71a0/bike.png"
          />
        </a>
        <a
          aria-current={null}
          href="/user"
          onClick={[Function]}
        >
          <img
            alt="ProfileMobile"
            className="nav__item"
            src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/e60b6c14d78fa03c5a1ae02b95f9cbca/user.png"
          />
        </a>
        <img
          alt="menu"
          className="nav__menu"
          onClick={[Function]}
          src="https://trello-attachments.s3.amazonaws.com/5f4cb639a6f5eb1005114de4/5f53d188f7827833bd79ef9b/ea9ccf0f7a5fe6c8d066bb89f204f504/open-menu.png"
        />
      </div>
    </div>
    <div
      className="header__desktop"
    >
      <img
        alt="logo"
        className="logo"
        src="https://cdn.discordapp.com/attachments/692420285143711814/693437226146594876/LogoGerili.png"
      />
      <p
        className="appName"
      >
        bikUP
      </p>
      <ul
        className="header__navigation"
      >
        <a
          aria-current={null}
          className="navigation__item"
          href="/bikes"
          onClick={[Function]}
        >
          My Bikes
        </a>
        <a
          aria-current={null}
          className="navigation__item"
          href="/search"
          onClick={[Function]}
        >
          Explore
        </a>
        <a
          aria-current={null}
          className="navigation__item"
          href="/user"
          onClick={[Function]}
        >
          Profile
        </a>
      </ul>
      <div
        className="flex-spacer"
      />
      <button
        className="login__button login__button--header"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </header>,
  <div
    className="general-container"
  >
    <div
      className="desktop"
    >
      <div
        className="standardaside"
      >
        <div
          className="standardaside__card-container"
        >
          <h3
            className="title"
          >
            Challenges
          </h3>
          <div
            className="standardaside__card"
          >
            <div>
              <div
                className="challenges__item"
              >
                <div
                  className="item__title"
                >
                  <img
                    alt="trophy"
                    className="title__img"
                    src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                  />
                  - 
                  BIrlfriend
                   -
                </div>
                <p>
                  Travel 5000km with a bike
                </p>
              </div>
              <div
                className="separator"
              />
            </div>
            <div>
              <div
                className="challenges__item"
              >
                <div
                  className="item__title"
                >
                  <img
                    alt="trophy"
                    className="title__img"
                    src="https://image.flaticon.com/icons/svg/3112/3112946.svg"
                  />
                  - 
                  CompoManiac
                   -
                </div>
                <p>
                  Change the same component 3 times
                </p>
              </div>
              <div
                className="separator"
              />
            </div>
          </div>
        </div>
        <div
          className="standardaside__card-container"
        >
          <h3
            className="title"
          >
            Give Away
          </h3>
          <div
            className="standardaside__card"
          >
            <p>
              something free
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="compodetail"
    >
      <div
        className="compodetail__upper mobile"
      >
        <a
          aria-current={null}
          href="/bikes/undefined"
          onClick={[Function]}
        >
          Back
        </a>
        <p
          className="upper__edit"
        >
          Edit
        </p>
      </div>
      <div
        className="compodetail__head"
      >
        <div
          className="head__container"
        >
          <a
            aria-current={null}
            className="desktop"
            href="/bikes/undefined"
            onClick={[Function]}
          >
            Back
          </a>
          <h2
            className="head__compoName"
          >
             -
             
          </h2>
          <button
            className="compodetail__reset desktop"
            onClick={[Function]}
          >
            Reset Component
          </button>
        </div>
        <div
          className="separator"
        />
        <div
          className="compodetail__km-hours"
        >
          <div
            className="km-hours km-hours__km"
          >
            <p
              className="km-hours__title"
            >
              Total KM
            </p>
            <div
              className="separator-small"
            />
            <p
              className="km__value"
            >
              NaN
            </p>
          </div>
          <div
            className="km-hours__hours km-hours"
          >
            <p
              className="km-hours__title"
            >
              Total Hours
            </p>
            <div
              className="separator-small"
            />
            <p
              className="km-hours__value"
            >
              NaN
            </p>
          </div>
        </div>
      </div>
      <div
        className="compodetail__components"
      >
        <div
          className="detailcompocard"
        >
          <div
            className="detailcompocard__content"
          >
            <img
              alt="compo img"
              className="detailcompocard__img"
              src="https://image.flaticon.com/icons/svg/2623/2623442.svg"
            />
            <div
              className="detailcompocard__info"
            >
              <div
                className="detailcompocard__labels"
              >
                <p />
                <p
                  className="labels__status"
                >
                  (
                  NaN
                   
                  /
                  NaN
                   Km)
                </p>
              </div>
              <div>
                <progress
                  id="progress"
                  max={NaN}
                  value={NaN}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="compodetail__details"
      >
        <div
          className="details__content"
        >
          <img
            alt="bike type"
            className="details__type-img"
            src="https://image.flaticon.com/icons/svg/2623/2623442.svg"
          />
          <div
            className="details__info"
          >
            <p
              className="info__type"
            >
               -
               
            </p>
            <p>
              Unknown brand
               
              -
               
              Unknown Model
            </p>
          </div>
        </div>
      </div>
      <button
        className="compodetail__reset mobile"
        onClick={[Function]}
      >
        Reset Component
      </button>
      <button
        className="compodetail__reset delete mobile"
        onClick={[Function]}
      >
        Delete Component
      </button>
    </div>
  </div>,
]
`;
